tests/infra/github.py

```diff
@@ -219,7 +219,7 @@ def get_latest_dev_tag(self):
 
     def get_tags_for_major_version(self, major_version=None):
         version_re = f"{major_version}\\." if major_version else ""
-        tag_re = f"^{TAG_RELEASE_PREFIX}{version_re}([.\\d+]+)(-rc.*|)$"
+        tag_re = f"^{TAG_RELEASE_PREFIX}{version_re}([.\\d+]+)$"
```

Comment:
Bundled with this PR, we also stop testing compatibility with release candidates as this was overly precautious, and we may break compatibility between release candidate and final release on purpose (just like we are doing for 4.0.0).

Reply:
I think it'd be better for clarity if this was split into its own PR.